===== Patch instructions generated by SAVER =====
[+] { Insert: if ({ p11_filter_allow_token:n$9 == { 0 } }) free(*(p11_filter_allow_token:token_copy)) at 9 (line 411, column 7) }
===== Actual patch in diff format =====
@@ -409,6 +409,7 @@ p11_filter_allow_token (p11_virtual *virt,
 	return_if_fail (token_copy != NULL);
 
 	if (!p11_array_push (filter->entries, token_copy))
+		free(token_copy);
 		return_if_reached ();
 
 	if (filter->initialized)
